### ISSUE TITLE
Device doesn't work normally with latest commits

### ISSUE THREAD

**Issue by nusbaoning** [OpenChannelSSD/qemu-nvme#17]
Hi. 
I have used the latest qemu-nvme (commit=6e5dab678c305f22a5f845156b570967910b3484) and the latest version of pblk.for-4.13 (commit=e9eaf4b94311ee7517199c31c4704f0fa4cacf3f) from https://github.com/OpenChannelSSD/linux/commits/pblk.for-4.13.

When I run fio with the following command
#sudo fio -filename=/dev/mydevice -direct=1 -iodepth 1 -thread -rw=randwrite -ioengine=psync -bs=16k -size=200M -numjobs=30 -runtime=10 -group_reporting -name=mytest. 
the fio task runs normally but the result is a little weird:
![image](https://user-images.githubusercontent.com/14070574/28563336-4068a45a-7158-11e7-9c7c-627830038360.png)
Since I set direct=1, the IO requests should bypass page cache and IO buffer so there is no reason that the disk did not receive any request.

I also tried to create ext4 and mount a partition, and copying and reading file on the device is successful.

However, when I tried another fio configuration:
[global]
ioengine=libaio
direct=1
thread=1
runtime=30
size=300M
filename=/dev/mydevice

[write4k-rand]
stonewall
group_reporting
bs=4k
rw=randwrite
numjobs=2
iodepth=4

The result is 
![image](https://user-images.githubusercontent.com/14070574/28563495-c250aa26-7158-11e7-8539-4e05bf40350f.png)
When I tried to create file system later, the dmesg showed:
![image](https://user-images.githubusercontent.com/14070574/28563552-e6d73c34-7158-11e7-9204-c1c50d6b03ae.png)
And this time, writing to the mounted folder is not allowed.
![image](https://user-images.githubusercontent.com/14070574/28563582-fca0163a-7158-11e7-9258-d10417729507.png)
The device can't be removed by nvme lnvm remove either.

Is the device working normally? Is my fio configuration wrong? Or any other problem?
Thank you so much!

**Comment by nusbaoning** [comment]
More experiment info.
The previous experiments were done on a real disk with ext4 file system.
I tried RAMdisk with ext2 FS and also a raw RAMdisk, all showed similar results.
One interesting thing is that the fio ioengine influenced the experiment result. If the ioengine is psync, the fio will stop normally but the result give disk states as all 0. However, if I run fio with ioengine libaio several times, it will give a input/output error and then the device can't be accessed normall, the fio even if with psync ioengine will run endlessly.

**Comment by nusbaoning** [comment]
Hi! I have tried to use the fio configuration given by javigon to run both on the ramdisk of my host machine and the ramdisk-emulated nvme device on the virtual machine.
[global]
ioengine=libaio
direct=1

[write]
filename=/dev/pblk0
norandommap
time_based=1
runtime=300
rw=randwrite
bs=256k
iodepth=1
numjobs=1

Both gave the result 
![image](https://user-images.githubusercontent.com/14070574/28608418-e3a4774c-7212-11e7-8278-5ff9f75f21d9.png)

What is the result when you run fio on an emulated nvme device? Is that possible that the device is working normally but the ramdisk and the emulated device do not return the disk states as fio wants?

For my previous error, the pblk generated block device is 352MB, and I set the size as 300MB in my fio configuration, any possibility that the size caused error? I am not familiar with the fio work mechanism. But the libaio engine is ok.
